acro/acro_stata_parser.py

```diff
@@ -0,0 +1,324 @@
+"""
+File with commands to manage the stata-acro interface
+Jim Smith 2023 @james.smith@uwe.ac.uk
+MIT licenses apply.
+"""
+import pandas as pd
+import statsmodels.iolib.summary as sm_iolib_summary
+
+from acro import ACRO, add_constant, stata_config
+from acro.utils import get_summary_dataframes, prettify_table_string
+
+
+def apply_stata_ifstmt(raw: str, all_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Parses an if statement from stata format
+    then uses it to subset a dataframe by contents.
+    """
+    if len(raw) == 0:
+        return all_data
+
+    # add braces aroubd each clause- keeping any in the original
+    raw = "( " + raw + ")"
+    raw = raw.replace("&", ") & (")
+    raw = raw.replace("|", ") | (")
+    # put spaces around operators to ease parsing
+    for operator in [">", "<", "==", ">=", "<=", "!="]:
+        raw = raw.replace(operator, " " + operator + " ")
+
+    # replace variable names with df["varname"]
+    for vname in all_data.columns:
+        raw = raw.replace(vname, 'all_data["' + vname + '"]')
+
+    # print(raw)
+    # apply exclusion
+    some_data = all_data[eval(raw)]  # pylint: disable=eval-used
```

Comment:
I'm not entirely sure about the difference in behaviour between `eval` and `ast.literal.eval` but if this could be changed to that here (needs import ast) and the functionality works the same then it would be nice to get rid of the warning on the static analysis report.

Reply:
fixed it by using df.query()
Added lots more tests and a few pragma:no cover around try...except cases

patch coverage is reporting la fraction of a percent low because I test the function that do stata subsetting separately rather than adding extra tests to do the same thing via an acro table command.
Will add a test to box those off later, but otherwise / notwithstanding, good to review I think